packages/apps/composer-app/src/plugins/welcome/credentials/util.ts

```diff
@@ -54,3 +54,56 @@ export const activateAccount = async ({
   const { credential } = await response.json();
   return credential && codec.decode<Credential>(credential);
 };
+
+type ProfileResponse = {
+  identityDid: string;
+  email?: string;
+  capabilities: string[];
+};
+
+/**
+ * Get profile.
+ *
+ * @param params.hubUrl
+ * @param params.credential
+ */
+export const getProfile = async ({
+  hubUrl,
+  credential,
+}: {
+  hubUrl: string;
+  credential: Credential;
+}): Promise<ProfileResponse> => {
+  const response = await fetch(new URL('/account/profile', hubUrl), {
+    headers: { Authorization: `Bearer ${codec.encode(credential)}` },
+  });
+  if (!response.ok) {
+    throw new Error('profile fetch failed', { cause: response.statusText });
```

Comment:
these are 400 errors, so bad requests, rather than exceptions? i'm not sure i'd do anything different here, but like the comment about it feels a bit leaky.

just something to consider when we evolve the hub-protocols client lib.